<launch>
<!-- rosmariokart -->
<node pkg="rosmariokart" type="rosmariokart.exe" name="rosmariokart"
      output="screen"
      >
<!--
      launch-prefix="valgrind"
-->
  <param name="axis_linear" value="3"/>
  <param name="axis_angular" value="0"/>
  <param name="axis_90turn" value="4"/>
  <param name="axis_180turn" value="5"/>
  <param name="scale_linear" value="0.7"/>
  <param name="scale_angular" value="2.5"/>
  <param name="button_item" value="3"/>
  <remap from="joy0" to="mip/joy"/>
  <remap from="joy1" to="sumo/joy"/>
  <remap from="sharp_turn0" to="mip/sharp_turn"/>
  <remap from="sharp_turn1" to="sumo/sharp_turn"/>
  <remap from="cmd_vel0" to="mip/cmd_vel"/>
  <remap from="cmd_vel1" to="sumo/cmd_vel"/>
</node>

<group ns="mip">
  <!-- Driver for first joystick. -->
  <node pkg="joy" type="joy_node" name="joy1" output="screen" >
    <param name="dev" value="/dev/input/js0" />
    <param name="autorepeat_rate" value="25" />
  </node>
  <!-- MIP driver -->
  <node pkg="rosmip" type="rosmip.exe" name="mip" respawn="false" output="screen"
        args="_device_mac:=00:19:0E:16:AF:22  _mip_mac:=D0:39:72:B7:AF:66
              _use_odometry_speed:=true" />
</group>

<group ns="sumo">
  <!-- Driver for second joystick. -->
  <node pkg="joy" type="joy_node" name="joy2" output="screen" >
    <param name="dev" value="/dev/input/js2" />
    <param name="autorepeat_rate" value="25" />
  </node>
  <!-- rossumo driver -->
  <node pkg="rossumo" type="rossumo.exe" name="sumo" respawn="false" output="screen">
    <param name="camera_calibration_filename" value="$(find rossumo)/data/sumo_camera_parameters.yaml"/>
    <param name="camera_calibration_camname" value="head_camera"/>
  </node>
</group>
</launch>
